frontend/server/src/main/java/org/pytorch/serve/grpcimpl/ManagementImpl.java

```diff
@@ -18,27 +19,48 @@
 import org.pytorch.serve.http.BadRequestException;
 import org.pytorch.serve.http.InternalServerException;
 import org.pytorch.serve.http.StatusResponse;
+import org.pytorch.serve.job.GRPCJob;
+import org.pytorch.serve.job.Job;
 import org.pytorch.serve.util.ApiUtils;
 import org.pytorch.serve.util.GRPCUtils;
 import org.pytorch.serve.util.JsonUtils;
+import org.pytorch.serve.util.messages.RequestInput;
+import org.pytorch.serve.wlm.ModelManager;
 
 public class ManagementImpl extends ManagementAPIsServiceImplBase {
 
     @Override
     public void describeModel(
             DescribeModelRequest request, StreamObserver<ManagementResponse> responseObserver) {
-
+        String requestId = UUID.randomUUID().toString();
+        RequestInput input = new RequestInput(requestId);
         String modelName = request.getModelName();
         String modelVersion = request.getModelVersion();
-
-        String resp;
-        try {
-            resp =
-                    JsonUtils.GSON_PRETTY.toJson(
-                            ApiUtils.getModelDescription(modelName, modelVersion));
-            sendResponse(responseObserver, resp);
-        } catch (ModelNotFoundException | ModelVersionNotFoundException e) {
-            sendErrorResponse(responseObserver, Status.NOT_FOUND, e);
+        boolean customized = request.getCustomized();
+
+        if ("all".equals(modelVersion) || !customized) {
```

Comment:
Not quite sure I understand this check? Can we not request a describe for a specific version of a model or was that never supported?

Reply:
existing describeAPI supports both a specific version and all versions. Extended describeAPI can only work for a specific version.